django/forms/widgets.py

```diff
@@ -181,6 +181,7 @@ class Widget(six.with_metaclass(MediaDefiningClass)):
     needs_multipart_form = False  # Determines does this widget need multipart form
     is_localized = False
     is_required = False
+    supports_microseconds = True
```

Comment:
I think separating it is difficult since there is no natural place to document `supports_microseconds` unless it is added at the base class level.

Regarding a test for 24497, isn't this already covered by the existing test: [test_initial_datetime_values](https://github.com/django/django/blob/master/tests/forms_tests/tests/test_forms.py#L1445) in particular the use of the `TextInput` widget at https://github.com/django/django/blob/master/tests/forms_tests/tests/test_forms.py#L1461?